aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -60,30 +60,12 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   if (self.is_contiguous(memory_format)) {
     return self;
   }
-  auto  result = at::empty_like(self);
-  switch (memory_format) {
-    case MemoryFormat::Any: // Back compatibility with old defaults
-    case MemoryFormat::Contiguous: {
-      break;
-    }
-    case MemoryFormat::ChannelsLast: {
-      TORCH_CHECK(
-          result.dim() == 4,
-          " required rank 4 tensor to use channels_last format");
-      std::vector<int64_t> newStrides(self.dim());
-      auto sizes = result.sizes();
-      newStrides[1] = 1;
-      newStrides[3] = sizes[1];
-      newStrides[2] = newStrides[3] * sizes[3];
-      newStrides[0] = newStrides[2] * sizes[2];
-      result = result.as_strided(sizes, newStrides);
-      break;
-    }
-    default: {
-      TORCH_CHECK(false, " unsupported memory format");
-    }
-  }
+  TORCH_CHECK(
+      memory_format != MemoryFormat::Preserve,
+      "preserve memory format is unsupported by the contiguous operator");
```

Comment:
Or we could just no-op if you pass preserve to contiguous :)  (I'm not sure if that makes any code easier to write lol)

Reply:
no-op meaning return itself? It is kind of unclear what to do with randomly permuted tensor in this case. I rather throw error instead of guessing what user mean.